hs_core/hydroshare/resource.py

```diff
@@ -722,16 +725,32 @@ def delete_resource_file(pk, filename_or_id, user):
 
     return filename_or_id
 
+def get_resource_doi(res_id, pending):
+    doi_str = "http://dx.doi.org/10.4211/hs.{shortkey}".format(shortkey=res_id)
+    if pending:
+        return "{doi}pending".format(doi=doi_str)
+    else:
+        return doi_str
+
+
+def get_activated_doi(doi):
+    idx = doi.find('pending')
+    if idx >= 0:
+        return doi[:idx]
+    else:
+        return doi
 
-def publish_resource(pk):
+def publish_resource(user, pk):
```

Comment:
@selimnairb @horsburgh @dtarb If there were an issue and Crossref gave a failed response, the failure details are returned as part of the response which is shown to users to inform them what is going on. During my testing using Crossref test system which is supposed to mimic exact functionality of their production system, I encountered a period of their database refresh, during which time CrossRef raised an error indicating user name xxx does not exist, and this error lasted for a few days before the system was back up running as expected. In this instance, periodic re-tries are a waste of resources which make things complicated since we have to decide what to do during this interim period before the error was fixed. As it is currently implemented, if there is any error with metadata deposit coming from CrossRef, an error is displayed on resource landing page indicating the details of the error raised from CrossRef, and the publication has failed with no effect to the resource. The user has to publish the resource again after the error is fixed. @selimnairb raised a good point, though. I do think we should better inform users of what has happened and what they should do next. I am proposing we expand the notification message in case there is an error raised from CrossRef with metadata deposit as the following: ```The resource publication has failed due to the following error raised from the DOI Registration Agency Crossref (http://www.crossref.org): <error_message_response_details_from_crossref>. If it is a transient network failure, you could try to publish this resource again. Otherwise, please contact <hydroshare support email> so that Crossref can be contacted to resolve this issue.``` @horsburgh @dtarb @selimnairb Please correct the message above as appropriate if you think this is what we should do.